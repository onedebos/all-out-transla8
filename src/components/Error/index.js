import styled from 'styled-components';

export default function Error({ error }) {
	return <ErrorStyles aria-label="error-text" data-testid="error-msg">{error}</ErrorStyles>;
}

const ErrorStyles = styled.p`
	color: #ddd;
	margin-top: 10px;
`;
