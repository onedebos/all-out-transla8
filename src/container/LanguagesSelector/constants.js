export default {
	srcLangOptions: [{ langCode: 'EN', lang: 'English' }],

	targetLangOptions: [
		{ langCode: 'BG', lang: 'Bulgarian' },
		{ langCode: 'CS', lang: 'Czech' },
		{ langCode: 'DA', lang: 'Danish' },
		{ langCode: 'DE', lang: 'German' },
		{ langCode: 'EL', lang: 'Greek' },
		{ langCode: 'EN-GB', lang: 'English (British)' },
		{ langCode: 'EN-US', lang: 'English (American)' },
		// "EN" - English (unspecified variant for backward compatibility; please select EN-GB or EN-US instead)
		{ langCode: 'ES', lang: 'Spanish' },
		{ langCode: 'ET', lang: 'Estonian' },
		{ langCode: 'FI', lang: 'Finnish' },
		{ langCode: 'FR', lang: 'French' },
		{ langCode: 'HU', lang: 'Hungarian' },
		{ langCode: 'IT', lang: 'Italian' },
		{ langCode: 'JA', lang: 'Japanese' },
		{ langCode: 'LT', lang: 'Lithuanian' },
		{ langCode: 'LV', lang: 'Latvian' },
		{ langCode: 'NL', lang: 'Dutch' },
		{ langCode: 'PL', lang: 'Polish' },
		{ langCode: 'PT-PT', lang: 'Portuguese' },
		{ langCode: 'PT-BR', lang: 'Portuguese (Brazilian)' },
		// {"PT" : Portuguese (unspecified variant for backward compatibility; please select PT:PT or PT:BR instead)}
		{ langCode: 'RO', lang: 'Romanian' },
		{ langCode: 'RU', lang: 'Russian' },
		{ langCode: 'SK', lang: 'Slovak' },
		{ langCode: 'SL', lang: 'Slovenian' },
		{ langCode: 'SV', lang: 'Swedish' },
		{ langCode: 'ZH', lang: 'Chinese' },
	],

	SRC_LANG: 'Source Language',
	TRGT_LANG: 'Target Language',
};
