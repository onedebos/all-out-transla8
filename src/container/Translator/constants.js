export default {
    NO_SRC_TEXT_ERR: "Enter a word to translate...",
    NO_TRGT_LANG_ERR: "Select a language to translate into."
}