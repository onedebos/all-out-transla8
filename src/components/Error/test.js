/**
 * @jest-environment jsdom
 */
 import { render, screen, cleanup } from '@testing-library/react';
 import '@testing-library/jest-dom';
 import renderer from 'react-test-renderer';
 import Error from './';
 import constants from '../../container/Translator/constants'
 
 afterEach(() => {
     cleanup();
 });
 
 describe('<Error />', () => {

    const {NO_SRC_TEXT_ERR} = constants

     it('renders the Button component successfully without crashing', () => {
         render(<Error error="" />);
     });

     it('renders the correct error text', () => {
         render(<Error error={NO_SRC_TEXT_ERR} />);
         const errorEl = screen.getByTestId('error-msg');
         expect(errorEl).toHaveTextContent(NO_SRC_TEXT_ERR);
     });
 
 
     it('matches snapshot', () => {
         const tree = renderer.create(<Error error={NO_SRC_TEXT_ERR} />).toJSON();
         expect(tree).toMatchSnapshot();
     });
 });
 