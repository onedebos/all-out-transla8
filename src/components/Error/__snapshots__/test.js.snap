// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> matches snapshot 1`] = `
<p
  aria-label="error-text"
  className="Error__ErrorStyles-sc-nms42m-0 bMSNJU"
  data-testid="error-msg"
>
  Enter a word to translate...
</p>
`;
